src/lib.rs

```diff
@@ -1,7 +1,34 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        assert_eq!(2 + 2, 4);
+mod caches;
+mod counters;
+mod entries;
+mod request;
+
+pub struct Infinispan {
+    base_url: String,
+    http_client: reqwest::Client,
+    basic_auth_encoded_val: String,
+}
+
+impl Infinispan {
+    pub fn new(
+        base_url: impl Into<String>,
+        username: impl AsRef<str>,
```

Comment:
Are username/password always required, or could be Option<>?

Later I guess you can add support for URL with username and password in the Url?

Reply:
They're required: https://infinispan.org/docs/stable/titles/server/server.html#endpoint_authentication